### ISSUE TITLE
Division of Labor

### ISSUE THREAD

**Issue by pdoehle** [pdoehle/math-robot#1]
Hey guy, I don't know if your interested in this project. I was thinking of targeting integration techniques first. I think we should have a clear vision for exactly what the software will do and then a clear division of labor so we're not tripping over each other again. What do you think about the project? Any interest?

**Comment by collinn** [comment]
Interesting...I'm not even sure where to start. But if this is meant to be an ongoing side project, I'm definitely interested. We are wanting to do this all in Python? Sweet.

**Comment by pdoehle** [comment]
Absolutely. Spend lots of time when you're interested, put it on the back burner when you're busy. Just something ongoing, mostly a way to learn python. I thought we could just begin first with a discussion here of what it should look like. I was thinking start small and then grow if it goes well. I'd like to practice some integration techniques. So I was thinking there could be like a general form for a specific kind of technique (like integration by parts for example). The program would dynamically generate a different problem of that type and then the tricky part will have it also be able to generate solution steps for whatever it randomly generates. For now I thought we could just use terminal notation (ie int[e^x]dx - or something like that. Maybe deciding on notation is a sub-problem that need to be addressed), but if it got good, then we could look at using some sort of web interface or with mathjax or latex. I figure start small and get the core functionality and worry about bells and whistles later if actually becomes something.

**Comment by pdoehle** [comment]
Notation I'm thinking right now:
+, -, =, /, *, !, cos, sin, etc  -> the usuals
d/dx[function here]  -> derivative
int[function here]dx  -> integral

Any big holes so far?

**Comment by pdoehle** [comment]
I'm also thinking if we had an engine that took a template ie: d/dx[rand/x]. Here the program would generate the derivative of a random number (maybe we need a way to specify range or int or something like that) over x. Then we can create a database of the templates for the different type of integration techniques and only have to program one engine. I want to start small (integration by parts), but design it so that its easy to scale it up for the other integration techniques and eventually other types of math problems as well.

**Comment by pdoehle** [comment]
Right now I'm thinking an initial division of labor could be one person does the generator and one person does the solver. That sounds unfair for the solver, but for now we're just focusing on integration by parts so it should be doable.
